### PR TITLE
[JBPM-9746] Precision time (milliseconds) for datime columns should be taken into account for MySQL and MariaDB

### PR DESCRIPTION
**Pull request by afalhambra**
**[JBPM-9746](https://issues.redhat.com/browse/JBPM-9746)**: Precision time (milliseconds) for datime columns should be taken into account for MySQL and MariaDB

### CODE REVIEW

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Missing lastModified update at BAMTaskSummary table for upgrade scripts.

Regarding task_assigning table upgrade, probably it's better not including it into test/ddl60 directory (as that table was not originally in that version, but created later, in 7.x). Default test shouldn't include it (as it's an optional table) and could be filtered out at `ScriptFilter.init()`. Then, a new test may be added to check the upgrade of this optional table if you want.

**Comment by afalhambra** [comment]
> Looks good to me, fantastic work @afalhambra !
> 
> Regarding task_assigning table upgrade, probably it's better not including it into test/ddl60 directory (as that table was not originally in that version, but created later, in 7.x). Default test shouldn't include it (as it's an optional table) and could be filtered out at `ScriptFilter.init()`. Then, a new test may be added to check the upgrade of this optional table if you want.

Thanks a lot @gmunozfe - I added your comments. Please have a look

File: jbpm-test-util/src/main/java/org/jbpm/test/persistence/scripts/TestPersistenceContextBase.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ public void executeScripts(final File scriptsRootFolder, ScriptFilter scriptFilt
     public void executeScripts(final File scriptsRootFolder, ScriptFilter scriptFilter,
                                DataSource dataSource, String defaultSchema) throws IOException, SQLException {
         final File[] sqlScripts = TestsUtil.getDDLScriptFilesByDatabaseType(scriptsRootFolder, databaseType, scriptFilter);
-        if (sqlScripts.length == 0 && scriptFilter.hasOption(Option.DISALLOW_EMTPY_RESULTS)) {
+        if (sqlScripts.length == 0 && scriptFilter.hasOption(Option.DISALLOW_EMPTY_RESULTS) && scriptFilter.isSupportedDatabase(databaseType)) {
```
Comment (gmunozfe) [inline jbpm-test-util/src/main/java/org/jbpm/test/persistence/scripts/TestPersistenceContextBase.java:144]: Is really needed this last part of the comparison?

**Review by gmunozfe — COMMENTED** [review]
Just a couple of questions and minor comments, amazing work @afalhambra !

File: jbpm-db-scripts/src/test/java/org/jbpm/persistence/scripts/UpgradeScriptsTest.java [hunk 2]
```diff
@@ -105,11 +108,32 @@ public void testExecutingScripts() throws IOException, SQLException {
         }
     }
 
+    /**
+     * Tests that DB schema is upgraded properly using database task_assigning_tables upgrade scripts.
+     * @throws IOException
```
[the remark this answers is not in the document]
Comment (afalhambra) [reply jbpm-db-scripts/src/test/java/org/jbpm/persistence/scripts/UpgradeScriptsTest.java:113]: done

File: jbpm-test-util/src/main/java/org/jbpm/test/persistence/scripts/TestPersistenceContextBase.java (hunk 1 above)
Comment (afalhambra) [reply jbpm-test-util/src/main/java/org/jbpm/test/persistence/scripts/TestPersistenceContextBase.java]: No, not really - removed.

**Comment by afalhambra** [comment]
Thanks for the review @gmunozfe - do you mind having a look now and let me know if that's ok? Thanks

File: jbpm-db-scripts/src/test/java/org/jbpm/persistence/scripts/UpgradeScriptsTest.java [hunk 3]
```diff
@@ -105,11 +108,39 @@ public void testExecutingScripts() throws IOException, SQLException {
...
+            ScriptFilter taskAssigningTable = filter("task_assigning_tables_")
+                                                .not("drop")
+                                                .setOptions(ScriptFilter.Option.DISALLOW_EMPTY_RESULTS, ScriptFilter.Option.THROW_ON_SCRIPT_ERROR);
+            executeScriptRunner(DB_DDL_SCRIPTS_RESOURCE_PATH, taskAssigningTable);
+            executeScriptRunner(DB_UPGRADE_SCRIPTS_RESOURCE_PATH, ScriptFilter.init(false, true).setDistribution(distributionType));
+            ScriptFilter scriptFilter = filter("task_assigning_tables.sql").
+                                        setDistribution(distributionType).
+                                        setOptions(ScriptFilter.Option.THROW_ON_SCRIPT_ERROR);
+            executeScriptRunner(DB_UPGRADE_SCRIPTS_RESOURCE_PATH, scriptFilter);
+            startAndPersistSomeProcess();
```
Comment (gmunozfe) [inline jbpm-db-scripts/src/test/java/org/jbpm/persistence/scripts/UpgradeScriptsTest.java:131]: Just one thought here: this test is just regression, nothing it's broken when upgrading the PlanningTask table, but there's no scenario using that table to check that it's working properly. wdyt?

**Review by gmunozfe — COMMENTED** [review]
A thought regarding upgrade test (functional) for task assigning table

**Comment by afalhambra** [comment]
Jenkins retest this

File: jbpm-db-scripts/src/test/java/org/jbpm/persistence/scripts/UpgradeScriptsTest.java (hunk 3 above)
Comment (afalhambra) [reply jbpm-db-scripts/src/test/java/org/jbpm/persistence/scripts/UpgradeScriptsTest.java:131]: As discussed earlier, it's not really feasible to test here as it's part of integration repo. So leaving it as is

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
not related failures
Jenkins retest this

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
@mareknovotny could you please merge?

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by afalhambra** [comment]
Jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by afalhambra** [comment]
Timed out - failures not related.
Jenkins retest this

**Comment by afalhambra** [comment]
Jenkins retest this

**Review by gmunozfe — APPROVED** [review]
Looks good to me, pitch-perfect work, well-done @afalhambra !

**Comment by afalhambra** [comment]
> Looks good to me, pitch-perfect work, well-done @afalhambra !

Thanks for the review @gmunozfe!

@mareknovotny could you please merge? thanks

**Pull request merged**